File: tarefa008/PessoaFisica.ts
import { Endereco } from "./Endereco.js"

export class PessoaFisica {
  constructor(nome:string, sexo:string, dataNascimento:Date) {
    this._nome = nome;
    this._sexo = sexo;
    this._dataNascimento = dataNascimento;
  }

  //atributos
  private _nome:string;
  private _sexo:string;
  private _dataNascimento:Date;
  private _endereco:Endereco;

  //setters
  set endereco(novoEndereco:Endereco) {
    this.endereco = novoEndereco;
  }
}

File: tarefa008/java/PessoaFisica.java
package tarefa008.java;

import java.sql.Date;

/**
 * Classe que representa a entidade Pessoa Física |
 * Autor: João Máiro |
 * Data: 15/07/22
 */
public class PessoaFisica {
  PessoaFisica(String nome, String sexo, Date dataNascimento) {
    this.nome = nome;
    this.sexo = sexo;
    this.dataNascimento = dataNascimento;
  }

  //atributos
  private String nome;
  private String sexo;
  private Date dataNascimento;
  private Endereco endereco;

  /**
   * Método que modifica o valor do atributo endereco
   * @param endereco
   */
  public void setEncereco(Endereco endereco) {
    this.endereco = endereco;
  }

  /**
   * Método que retorna o valor do atributo nome
   * @return
   */
  public String getNome() {
    return this.nome;
  }

  /**
   * Método que retorna o valor do atributo sexo
   * @return
   */
  public String getSexo() {
    return this.sexo;
  }

  /**
   * Método que retorna o valor do atributo dataNascimento
   * @return
   */
  public Date getDataNascimento() {
    return this.dataNascimento;
  }

  /**
   * Método que retorna o valor do atributo endereco
   * @return
   */
  public Endereco getEndereco() {
    return this.endereco;
  }
}
